Kernel/Process.cpp

```diff
@@ -2636,6 +2636,21 @@ int Process::sys$systrace(pid_t pid)
     return fd;
 }
 
+int Process::sys$halt()
+{
+    if (!is_superuser())
+        return -EPERM;
+
+    dbgprintf("acquiring FS locks...\n");
+    FS::lock_all();
+    dbgprintf("syncing mounted filesystems...\n");
+    FS::sync();
+    dbgprintf("attempting system shutdown...\n");
```

Comment:
Not important, but it'd be nice to leave the "bye, friends" thing in there I think. I put that wording in there kind of deliberately due to @awesomekling's videos always starting with "Hello, friends!" :)

Reply:
Haha I'll add it back in tomorrow in another PR. I think I deleted it bu accident right by I pushed (I had to leave the house) so I just wrote something boring and technical instead :P